src/pages/1_analysis.py

```
@@ -0,0 +1,288 @@
+) -> Optional[Tuple[np.ndarray, np.ndarray, str]]:
+    """アップロードされたExcelファイルからデータを抽出する"""
+    try:
+        df = pd.read_excel(uploaded_file, header=None)
+        if df.shape[1] < 2:
+            st.error("Excelファイルには少なくとも2つの列（時間と値）が必要です。")
+            return None
+        df = df.iloc[:, :2].apply(pd.to_numeric, errors="coerce").dropna()
+        time_array = df.iloc[:, 0].values
+        value_array = df.iloc[:, 1].values
+        filename = os.path.splitext(uploaded_file.name)[0]
+        return time_array, value_array, filename
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)